test/mailers/user_mailer_test.rb

```diff
@@ -10,135 +10,173 @@ def default_url_options
+
+    email = UserMailer.competition_reminder(@first_place_user).deliver_now
 
     assert_not ActionMailer::Base.deliveries.empty?
 
-    assert_equal ['do-not-reply@mitrecyberacademy.org'], email.from
-    assert_equal ['mitrectf+user1test@gmail.com'], email.to
-    assert_equal 'MITRE CTF: Competition Reminder', email.subject
-    assert_equal @remind_email_body, strip_tags(email.body.to_s).squish
+    assert_equal [@game.do_not_reply_email], email.from
+    assert_equal [@first_place_user.email], email.to
+    assert_equal "#{@game.title}: Competition Reminder", email.subject
   end
 
   test 'ranking no employment' do
-    user = users(:user_one)
-    email = UserMailer.ranking(user).deliver_now
+    email = UserMailer.ranking(@second_place_user).deliver_now
+
+    assert_not ActionMailer::Base.deliveries.empty?
+    assert_equal [@game.do_not_reply_email], email.from
+    assert_equal [@second_place_user.email], email.to
+    assert_equal "#{@game.title}: Congratulations!", email.subject
+    assert_equal true, email.has_attachments?
+  end
+
+  test 'ranking with user interested in employment and game having no employment information' do
+    @second_place_user.update(interested_in_employment: true)
+
+    email = UserMailer.ranking(@second_place_user).deliver_now
+
+    assert_not ActionMailer::Base.deliveries.empty?
+    assert_equal [@game.do_not_reply_email], email.from
+    assert_equal [@second_place_user.email], email.to
+    assert_equal "#{@game.title}: Congratulations!", email.subject
+    assert_equal true, email.has_attachments?
+  end
+
+  test 'ranking with user interested in employment and game having employment information' do
+    @second_place_user.update(interested_in_employment: true)
+    @game.update(participant_recruitment_url: Faker::Internet.url(host: 'mitre.org'))
+
+    email = UserMailer.ranking(@second_place_user).deliver_now
+
+    assert_not ActionMailer::Base.deliveries.empty?
+    assert_equal [@game.do_not_reply_email], email.from
+    assert_equal [@second_place_user.email], email.to
+    assert_equal "#{@game.title}: Congratulations!", email.subject
+    assert_equal true, email.has_attachments?
+  end
+
+  test 'ranking email for first place with scholarships available' do
+    @game.update(scholarships_available: true)
+    email = UserMailer.ranking(@first_place_user).deliver_now
 
     assert_not ActionMailer::Base.deliveries.empty?
 
-    assert_equal ['do-not-reply@mitrecyberacademy.org'], email.from
-    assert_equal ['mitrectf+user1test@gmail.com'], email.to
-    assert_equal 'MITRE CTF: Congratulations!', email.subject
+    assert_equal [@game.do_not_reply_email], email.from
+    assert_equal [@first_place_user.email], email.to
+    assert_equal "#{@game.title}: Congratulations!", email.subject
     assert_equal true, email.has_attachments?
-    assert_includes (strip_tags(email.body.parts.first.to_s).squish.to_s), @rank_email_body
```

Comment:
Can we still assert some basic checks here such as 'Your team came ranked x'? Otherwise having tests for every email use case is pretty useless since we can't verify that ranking is working correctly based on the known ranks in the tests and breaking variable interpolation in the emails would not raise any errors.